terraform/context.go

```diff
@@ -253,7 +261,7 @@ func (c *Context) Apply() (*State, error) {
 //
 // Plan also updates the diff of this context to be the diff generated
 // by the plan, so Apply can be called after.
-func (c *Context) Plan(opts *PlanOpts) (*Plan, error) {
+func (c *Context) Plan() (*Plan, error) {
```

Comment:
Important core note here - `Destroy` (whether or not we're doing a `terraform destroy` or `terraform plan -destroy`) is moving from a property of the `Plan` to a property of the `Context`.

This is so that `Destroy` can be used to switch `Transform` behavior, which makes Target calculation much simpler than if we had to do it later in the pipeline.

It was the only runtime input to `Plan()` so `PlanOpts` goes away without it.